### DISCUSSION TITLE
does cleaning data from kafka can make milvus lose some data

### DISCUSSION THREAD

**Discussion by marait123** [milvus-io/milvus#22480] [Q&A and General discussion] [answered]
hello I am using milvus with the following specification
```
mode: cluster mode
log broker: kafka (disk=25GB, retention_time=160hours)
```
currently we have an issue with kafka disk space `java.io.IOException: No space left on device` after connecting milvus to kafka and inserting around (100,000) vectors inside of milvus.

we are currently planning to solve this by reducing the log retention time but we are afraid this might cause the system to be in inconsistent state since milvus uses kafka logs to store it's redo-log which it uses for keeping all the data nodes in consistent state.

do you have any idea if it is safe to decrease the log retention time or not ? thank you.

**Comment by xiaofan-luan** [comment, marked as answer]
Hi @marait123 
Change Retention time to 1 day could probably help.
Milvus generate checkpoint every a few minutes. So as long as checkpoint works kafka data is no longer needed.

The only risk we don't want to make retention super small is Milvus crashing without doing checkpoint for long time. Otherwise change  retention time smaller works fine

**Reply by marait123** [reply]
Hi @xiaofan-luan 
thnaks for you answer.
but I read about sealed segments and growing segments , and a segment is flushed after it is sealed. but this where are growing segments being stored ?

**Reply by marait123** [reply]
I doubt that if a crash happens they could restored from checkpoints saved in minio?

**Reply by xiaofan-luan** [reply]
Yes, checkpoint also works for growing segment.
It's actually two different operation:
1. Sync: save in memory data to Minio and save checkpoint 
2. Seal: close growing segment and disable further writing (For index build)

Sync would be good enough for generating checkpoint and it will happen every few minutes(or if insert data size larger than 64M)

**Comment by farioas** [comment]
How often does the Milvus make a checkpoint? 
Can I set Kafka retention time to 1hr?

**Reply by yhmo** [reply]
Time tick interval is defined in milvus.yaml, default value is 200ms
```
proxy:
  timeTickInterval: 200 # ms, the interval that proxy synchronize the time tick
```

Kafka/Pular is like a WAL component for milvus. All the inserted data will be firstly written into Kafka/Pulsar before being consumed by Milvus data node and query node. Sometimes if you have intensive insert actions, the consume speed of datanode/querynode is slower than insert speed, then it is risky that data lost if you set Kafka/Pulsar retention time to a short value.
